packages/cli/src/concurrency/concurrency-control.service.ts

```diff
@@ -89,16 +108,24 @@ export class ConcurrencyControlService {
 	async throttle({ mode, executionId }: { mode: ExecutionMode; executionId: string }) {
 		if (!this.isEnabled || this.isUnlimited(mode)) return;
 
-		await this.productionQueue.enqueue(executionId);
+		const queue = this.getQueue(mode);
+
+		if (queue) {
+			await queue.enqueue(executionId);
+		}
```

Comment:
```suggestion
		await this.getQueue(mode)?.enqueue(executionId);
```

Same for others below